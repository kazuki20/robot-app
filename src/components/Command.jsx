import React from "react";
import { Row, Col, Button } from "reactstrap";
import compass from "../img/compass.png";

export default function Command({
  report,
  move,
  left,
  right,
  isRobotPlaced,
  toggleModalPlaceInput
}) {
  return (
    <div
      style={{
        height: "90%",
        border: "2px solid",
        textAlign: "center"
      }}
    >
      <Row className="mb-4 mt-4">
        <Col>
          <img src={compass} alt="Compass" style={{height: "20vh"}}/>
        </Col>
      </Row>
      <Row className="mb-4 mt-4">
        <Col>
          <Button
            color="primary"
            className="mr-2"
            onClick={toggleModalPlaceInput}
            name="place"
          >
            Place
          </Button>
          <Button
            color="primary"
            className="ml-2"
            onClick={report}
            disabled={isRobotPlaced ? false : true}
            name="report"
          >
            Report
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <Button
            color="primary"
            onClick={move}
            disabled={isRobotPlaced ? false : true}
            name="move"
          >
            Move ü†ù
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <Button
            color="primary"
            onClick={left}
            disabled={isRobotPlaced ? false : true}
            name="left"
          >
            ‚ü≤ Left
          </Button>
        </Col>
        <Col>
          <Button
            color="primary"
            onClick={right}
            disabled={isRobotPlaced ? false : true}
            name="right"
          >
            Right ‚ü≥
          </Button>
        </Col>
      </Row>
    </div>
  );
}
