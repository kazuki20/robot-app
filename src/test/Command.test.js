import React from 'react';
import { shallow } from 'enzyme';
import Command from '../components/Command';
import { Button } from "reactstrap";

function createTestProps(props) {
  return {
    isRobotPlaced: false,
    ...props,
  };
}

let wrapper, props;

beforeEach(() => {
  props = createTestProps();
  wrapper = shallow(<Command {...props} />);
});

describe('<Command /> rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 5 <Button /> s to display the Place, Report, Move, Left and Right options', () => {
    expect(wrapper.find(Button)).toHaveLength(5);
  });

  it('should render one enable button when isRobotPlaced is false', () => {
    wrapper = shallow(<Command isRobotPlaced={false} />);
    expect(wrapper.find("Button").not("Button[disabled=true]")).toHaveLength(1);
  });
});

describe('<Command /> interactions', () => {
  it('should invoke the toggleModalPlaceInput callback when Button Place is clicked', () => {
    let mockFn = jest.fn();

    let wrapper = shallow(<Command toggleModalPlaceInput={mockFn} />);
    wrapper.find('Button[name="place"]').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should invoke the report callback when Button Place is clicked', () => {
    let mockFn = jest.fn();

    let wrapper = shallow(<Command report={mockFn} />);
    wrapper.find('Button[name="report"]').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should invoke the move callback when Button Place is clicked', () => {
    let mockFn = jest.fn();

    let wrapper = shallow(<Command move={mockFn} />);
    wrapper.find('Button[name="move"]').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should invoke the left callback when Button Place is clicked', () => {
    let mockFn = jest.fn();

    let wrapper = shallow(<Command left={mockFn} />);
    wrapper.find('Button[name="left"]').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should invoke the right callback when Button Place is clicked', () => {
    let mockFn = jest.fn();

    let wrapper = shallow(<Command right={mockFn} />);
    wrapper.find('Button[name="right"]').simulate('click');

    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});