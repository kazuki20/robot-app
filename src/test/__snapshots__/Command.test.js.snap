// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Command /> rendering renders correctly 1`] = `
<div
  style={
    Object {
      "border": "2px solid",
      "height": "90%",
      "textAlign": "center",
    }
  }
>
  <Row
    className="mb-4 mt-4"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <img
        alt="Compass"
        src="test-file-stub"
        style={
          Object {
            "height": "20vh",
          }
        }
      />
    </Col>
  </Row>
  <Row
    className="mb-4 mt-4"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        className="mr-2"
        color="primary"
        name="place"
        tag="button"
      >
        Place
      </Button>
      <Button
        className="ml-2"
        color="primary"
        disabled={true}
        name="report"
        tag="button"
      >
        Report
      </Button>
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="primary"
        disabled={true}
        name="move"
        tag="button"
      >
        Move ü†ù
      </Button>
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="primary"
        disabled={true}
        name="left"
        tag="button"
      >
        ‚ü≤ Left
      </Button>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="primary"
        disabled={true}
        name="right"
        tag="button"
      >
        Right ‚ü≥
      </Button>
    </Col>
  </Row>
</div>
`;
